app/models/service.rb

```diff
@@ -85,8 +85,47 @@ def deployment_finished?(deployment_id)
     backend.deployment_finished?(deployment_id)
   end
 
+  def save_and_update_container_count!(desired_container_count)
+    self.desired_container_count = desired_container_count
+    self.save!
+
+    ecs.update_service({
+      desired_count: desired_container_count,
+      cluster: district.name,
+      service: logical_name
+    })
+  end
+
+  def logical_name
+    arn.split('/').last
+  end
+
+  def arn
+    service_arns.find { |x| x.start_with? arn_prefix }
```

Comment:
Question: do `service_arns` always have at least one arn starting with the prefix? Otherwise `logical_name` can raise `NoMethodError`